src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
-    @Override
-    public boolean add(T value) {
-        return false;
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
+    public MyLinkedList() {
+    }
+
+    private static class Node<T> {
+        private T data;
+        private Node<T> next;
+        private Node<T> previous;
+
+        public Node(Node<T> previous, T data, Node<T> next) {
+            this.data = data;
+            this.next = next;
+            this.previous = previous;
+        }
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean add(T data) {
+        insertLast(data);
+        return true;
+    }
 
+    @Override
+    public void add(T data, int index) {
+        checkBounds(index);
+        if (index == size) {
+            insertLast(data);
+        } else {
+            insertBefore(data, getNode(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        T[] array = (T[])list.toArray();
```

Comment:
do we need to create new array or we can iterate list from the input?

Reply:
fixed!